test/higher_order_ops/test_invoke_subgraph.py

```diff
@@ -244,19 +255,25 @@ def gn(x, y):
         def fn(x, y):
             nonlocal counter
             counter = 2
-            a = invoke_subgraph(gn, None, (x, y))[0]
+            a = invoke_subgraph(gn, (x, y), identifier=None)[0]
             counter = 3
-            return invoke_subgraph(gn, None, (a, y))[0]
+            return invoke_subgraph(gn, (a, y), identifier=None)[0]
 
-        x = torch.randn(8, requires_grad=True)
-        y = torch.randn(8, requires_grad=True)
-        ref = fn(x, y)
```

Comment:
gn has a free variable in it (counter). This breaks invoke_subgraph in eager (for some of the implementations I tried). Instead, I end up testing fn with compile vs fn2 (without invoke subgraph) with eager.

Our wrapper invoke_subgraph function should say something like "if you're compiling, call invoke_subgraph, otherwise, do eager".